### ISSUE TITLE
Support for .NET Core

### ISSUE THREAD

**Issue by jjeffery** [aws/aws-sdk-net#214]
Are there any plans for the AWS SDK for .NET to support the new .NET Core version? This is the .NET version that will run on Windows, OSX and Linux.

**Comment by normj** [comment]
The new Core CLR is very exciting and is on our radar. I think it will fit in nicely with the new modularized version 3 of the SDK. I'll leave this open and post back when I have more information.

**Comment by salerth** [comment]
In the interim, could you look at adding the dnxcore50 tags to the nuget packages?  The portable libs work fine under DNX (even though you only get the Async methods), however DNU/DNX doesn't realize you can use them.  When you run a `dnu restore` it picks up the net45core dependencies just fine, but it doesnt resolve the `dnxcore50` dependencies.

I'm still pretty new to this so there might be a better way, but right now if I tweak my project.lock.json to tell it to use the portables it works and compiles.

```
"AWSSDK/2.3.42.0": {
        "dependencies": {
          "Microsoft.Net.Http": "2.1.10"
        }
        "compile": {
          "lib/windows8/AWSSDK.dll": {}
        },
        "runtime": {
          "lib/windows8/AWSSDK.dll": {}
        }
      }
```

**Comment by normj** [comment]
I'm surprised you got the the existing PCL to work for CoreCLR. I was under impression that would not work. I'll need to do some testing on my end first before adding the dnxcore50 tag.

From my research CoreCLR only supports HttpClient with async only so no matter what it looks like CoreCLR support will be async only.

**Comment by salerth** [comment]
I'll admit I've only tried one method so far, but on OSX I can call DescribeRegionsAsync and it returns properly.

**Comment by danielcrenna** [comment]
Any hints on how to build from source? It currently does not. 

I was hoping to try my luck with a dnxcore50 port of the portable projects, but master does not build due to various missing interfaces.

**Comment by INRIX-dion-hutchings** [comment]
+1 Yes please add DNXCore support.  Thanks.

**Comment by normj** [comment]
Daniel

How are you trying to build the portable class library. If you are building from the solution file then you probably need to right click on the solution and "Enable nuget restore". The solution files are created as part of the generator, we need to get the generator to have "enable nuget restore" setup by default.

**Comment by adamhathcock** [comment]
Is there a shared branch with the work mentioned in this issue?  I'm interested in helping out.

**Comment by normj** [comment]
There is not a branch setup yet for this work. We have some upcoming project restructuring changes coming to the V3 work. After that gets pushed I'd like to start a development branch for dnxcore.

**Comment by normj** [comment]
@salerth I believe the reason you are seeing things work when you add the dnxcore50 tag in the nuget package is because your current dnx runtime is still set for clr. If you switch your dnx runtime to coreclr by doing "dnvm upgrade -r coreclr" then you will fail. That is because the SDK is currently referencing types that haven't port to dnxcore yet.

I'm thinking what we need to do is create another skew of the Core package that will get included in the nuget package like we have the .NET 4.5 and .NET 3.5.

**Comment by salerth** [comment]
@normj Yes you're probably right, DNX on OSX/Linux is pretty confusing.  I have to use the `mono` runtime (essentially `clr` as you say I guess) as `coreclr` doesn't work yet outside of windows so I can't really test this... I'll fire up a windows box later to play around (I should anyway as I've never really tried my app on windows/visual studio).  

I guess I'm happy with the hack above as the most important thing for me currently is to ensure my application will be compatible with the coreclr (when it is complete) rather than using it right now.

In summary, dnx confuses me ;)

**Comment by tmitchel2** [comment]
+1 for CoreClr support.  I'm currently using it with Clr (Mono Framework) but really would like to switch!!!

**Comment by adamhathcock** [comment]
Now that the v3 SDK is out (which is nice to have separate packages, btw) any chance of starting a DNX/CoreCLR branch?

I'm happy to try myself though I need to familiarize with the code generation and I'd rather have someone else point me to what needs doing.

**Comment by normj** [comment]
Yes now that V3 is out I'm working on CoreCLR support. I have a development branch going that I plan on pushing to GitHub soon. I want to make sure that our build scripts are working with CoreCLR before I push the branch so anybody that wants to contribute could run the build scripts to create NuGet packages.

**Comment by adamhathcock** [comment]
Thanks, that's great.  I wasn't trying to pester, just offer :)

**Comment by normj** [comment]
Closing this in favor of the new issue I opened up for tracking the support which is currently in development on the dnxcore-development.

https://github.com/aws/aws-sdk-net/issues/229
